### ISSUE TITLE
Two classes with same name but different name space are documented as one

### ISSUE THREAD

**Issue by blochberger** [realm/jazzy#999]
I have a class `Foo` which is public and a private helper class of the same, but as part of a subclass. The documentation of both classes is merged into one, e.g.:

```swift
class Foo {
    let a = "a"
}

class Helpers {
    class Foo {
        let b = "b"
    }
}
```

The documentation will list `a` and `b` under a single class `Foo`. Namespaces should be respected.

**Comment by johnfairh** [comment]
I realise this is very old now, but I think jazzy gets this right.

The example above gives me two separate files, `docs/Classes/Foo.html` and `docs/Classes/Helpers/Foo.html` with the right contents.  I've played around with access level, source file and not got it to break.  I had a quick go at your linked project's prior commit and I *think* the docs look fine (on current jazzy + Xcode of course which is probably not what you were doing back in the summer.)

Jazzy does merge declarations in some scenarios, but that should give a CLI error if extensions aren't involved or if there are compile errors confusing things.  Even then, merging decls with different nesting levels looks very hard to reach.

**Comment by blochberger** [comment]
I just tried to reproduce the issue with the code snipped pasted above with the most recent version (ruby-2.6.0.p0 with jazzy-0.9.4). The nested class `Helpers.Foo` is listed below `Helpers` which is as expected. I also tried generating the documentation of https://github.com/blochberger/Tafelsalz/commit/fa3dca1f245bbbb11982b836d80de82d02f03a23 with the `build-docs.sh` script and the class `GenericHash` was not merged as was previously the case. Hence I assume that it got solved and close the issue.
